### PR TITLE
refactor: #65 Use flag instead of aray[0]

### PR DESCRIPTION
**Pull request by Howl404**

### CODE REVIEW

File: mw-webapp/src/component/table/columns.tsx [hunk 1]
```diff
@@ -22,14 +22,13 @@ const getObjectArrayItem = (arrayItem: JobDone | PlanForNextPeriod | CurrentProb
   );
 };
 
-const getStringArrayItem = (arrayItem: string, index: string) => {
+const getStringArrayItem = (arrayItem: string, index: string, isDone?:boolean) => {
```
Comment (tritonJS826) [inline mw-webapp/src/component/table/columns.tsx:25]: 1. weird function name, rename it or at least leave comment please) I see it not youth but let's make our code better) 
2. usually if want to declare function with few parameters it is a good idea to use object as singular parameter. In this case your parameters become named object fields, you will be able to create a model for your parameters. Also It is pretty scalable solution.
3. if some function return jsx - let's name it with prefix `render*`. Could you add it to conventions, please?

File: mw-webapp/src/model/businessModel/DayReport.ts [hunk 2]
```diff
@@ -33,9 +34,9 @@ export class DayReport {
   public problemsForCurrentPeriod: CurrentProblem[];
 
   /**
-   * Anything that student wants to say about work
+   * Array of student comments and their status
```
Comment (tritonJS826) [inline mw-webapp/src/model/businessModel/DayReport.ts:37]: Dont duplicate information it will be hard to maintain it. What if the `StudentComment` model changed or the status field changed etc.. Document only useful information that couldn't be placed in code (or if it is hard etc.).
in current case `Student comment` will be enough (yeah code is self-documented in this case)

File: mw-webapp/src/model/businessModel/MonthReport.ts [hunk 3]
```diff
@@ -34,9 +35,9 @@ export class MonthReport {
   public problemsForCurrentPeriod: CurrentProblem[];
 
   /**
-   * Anything that student wants to say about work
+   * Array of student comments and their status
```
Comment (tritonJS826) [inline mw-webapp/src/model/businessModel/MonthReport.ts:38]: you have typescript to handle types, and fields name don't use docs for that

File: mw-webapp/src/model/businessModel/StudentComment.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Student comments
+ */
+export class StudentComment {
```
Comment (tritonJS826) [inline mw-webapp/src/model/businessModel/StudentComment.ts:4]: Is it one comment or few comments? Mistake in class name or in the documentation

File: mw-webapp/src/model/firebaseCollection/DayReport.ts [hunk 5]
```diff
@@ -29,9 +31,9 @@ export class DayReport {
   public problemsForCurrentPeriod: string[];
 
   /**
-   * Anything that student wants to say about work
+   * Array of student comments and their status
```
Comment (tritonJS826) [inline mw-webapp/src/model/firebaseCollection/DayReport.ts:34]: Array, status - the same - not fo rdocs

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/model/businessModel/StudentComment.ts (hunk 4 above)
Comment (Howl404) [reply mw-webapp/src/model/businessModel/StudentComment.ts:4]: Done

File: mw-webapp/src/model/businessModel/MonthReport.ts (hunk 3 above)
Comment (Howl404) [reply mw-webapp/src/model/businessModel/MonthReport.ts:38]: Done

File: mw-webapp/src/model/firebaseCollection/DayReport.ts (hunk 5 above)
Comment (Howl404) [reply mw-webapp/src/model/firebaseCollection/DayReport.ts:34]: Done

File: mw-webapp/src/model/businessModel/DayReport.ts (hunk 2 above)
Comment (Howl404) [reply mw-webapp/src/model/businessModel/DayReport.ts]: Done

File: mw-webapp/src/component/table/columns.tsx (hunk 1 above)
Comment (Howl404) [reply mw-webapp/src/component/table/columns.tsx:25]: 1. Problem is, this function is used by: stunderComments, mentorComments and learnedForToday, i can't really think of the more specific name for it. Maybe just "renderStringItem"

2. Considering the above-mentioned, if i understood correctly, i would need to create StringItem only in Business models?

3. Will add to conventions.

File: mw-webapp/src/component/table/columns.tsx [hunk 6]
```diff
@@ -99,13 +98,14 @@ export const columns: ColumnDef<DayReport, Date & JobDone[] & PlanForNextPeriod[
...
+  columnHelper.accessor<"studentComments", StudentComment[]>("studentComments", {
     header: "Student comments",
     cell: ({row}) => {
       const parentID = row.original.uuid;
+
       return (
         row.original.studentComments
-          ?.map((studentCommentItem) => (getStringArrayItem(studentCommentItem, parentID)))
+          ?.map((studentCommentItem) => (getStringArrayItem(studentCommentItem.description, parentID, studentCommentItem.isDone)))
       );
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/table/columns.tsx:109]: If you changed model for student comments then you shouldn't use function getStringArrayItem. There exists function getObjectArrayItem for cells  that have custom type

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/model/firebaseCollection/StudentComment.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+  /**
+   * True if comment was done and false if not
+   */
+  public isDone: boolean;
+
+  constructor(studentCommentData: StudentComment) {
+    this.description = studentCommentData.description;
+    this.isDone = studentCommentData.isDone;
+  }
```
[the remark this answers is not in the document]
Comment (Howl404) [reply mw-webapp/src/model/firebaseCollection/StudentComment.ts:19]: So does we need a option to mark simple comments as done or not? I'm just refactoring the existing functionality

File: mw-webapp/src/component/table/columns.tsx (hunk 6 above)
Comment (Howl404) [reply mw-webapp/src/component/table/columns.tsx:109]: 1. StudentComments don't have required uuid
2. I would need to change whole function to correctly display it

File: mw-webapp/src/model/firebaseCollection/StudentComment.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/model/firebaseCollection/StudentComment.ts:19]: But this cells will always contain simple comments

File: mw-webapp/src/model/firebaseCollection/DayReport.ts [hunk 8]
```diff
@@ -29,9 +31,9 @@ export class DayReport {
   public problemsForCurrentPeriod: string[];
 
   /**
-   * Anything that student wants to say about work
+   * Student comments
    */
-  public studentComments: string[];
```
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/model/firebaseCollection/DayReport.ts:34]: Victor told don't repeat info. He was mean don't write types of properties (arrays, strings etc) But I say about explain what is studentComment 
@tritonJS826 Do I understand wrong your position?

File: mw-webapp/src/component/table/columns.tsx (hunk 6 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/table/columns.tsx:109]: Right? student comment hasn't uuid because it was just string[] not custom model

File: mw-webapp/src/model/firebaseCollection/StudentComment.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (Howl404) [reply mw-webapp/src/model/firebaseCollection/StudentComment.ts:19]: So we don't need a option to mark it as done \ completed?

File: mw-webapp/src/model/firebaseCollection/DayReport.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (Howl404) [reply mw-webapp/src/model/firebaseCollection/DayReport.ts:34]: "Document only useful information that couldn't be placed in code (or if it is hard etc.).
in current case Student comment will be enough (yeah code is self-documented in this case)"

File: mw-webapp/src/model/firebaseCollection/StudentComment.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/model/firebaseCollection/StudentComment.ts:19]: Give please example when studentComment can be with property isDone: false

File: mw-webapp/src/component/table/columns.tsx (hunk 6 above)
Comment (Howl404) [reply mw-webapp/src/component/table/columns.tsx:109]: Yes and you told me student comment will not need uuid. So should i completely change a function because of one model?

File: mw-webapp/src/model/firebaseCollection/StudentComment.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (Howl404) [reply mw-webapp/src/model/firebaseCollection/StudentComment.ts:19]: Maybe i'm not understanding the question correctly but in this context studentComment should have isDone: false by default

[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/model/firebaseCollection/StudentComment.ts:19]: Why it should be false by default? 
Give example of real student comment. (I mean image that student wrote it in the cell)/ So give some examples. what the student can write in this sell with property  isDone: false

For example if student will write ""It was hard to read 100 pages of MsConel" it can't be false

File: mw-webapp/src/component/table/columns.tsx (hunk 6 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/table/columns.tsx:109]: So I think that studentComment no need uuid because I think that if studentComments is a string[] it's enough

But you changed string[] to custom model.

Comment (Ekaterina1994) [reply mw-webapp/src/component/table/columns.tsx:109]: So I think that studentComment no need uuid because I think that if studentComments is a string[] it's enough

But you changed string[] to custom model.

File: mw-webapp/src/model/firebaseCollection/DayReport.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/model/firebaseCollection/DayReport.ts:34]: It's good if everyone understand thr same what you student can write in the cell studendComments/ But even now we have long discuss because of misunderstanding. So without explain what is inside studentComment discuss will be in the future too

File: mw-webapp/src/component/table/columns.tsx (hunk 1 above)
Comment (tritonJS826) [reply mw-webapp/src/component/table/columns.tsx:24]: 1. Provide appropriate comment for it
2. No, I mean signature of this function itself, instead of 3 parameters use only one - object
3. Super) 👍

File: mw-webapp/src/component/table/columns.tsx (hunk 6 above)
Comment (tritonJS826) [reply mw-webapp/src/component/table/columns.tsx]: This table component is really bad, also it is not a component (read docs), don't hesitate to change anything if it could improve coffee

**Review by tritonJS826 — COMMENTED** [review]
Almost done)

**Comment by Howl404** [comment]
I'm almost done but probably better to wait for #78 merge, and refactor my code

**Comment by tritonJS826** [comment]
No problem)

File: mw-webapp/src/model/firebaseCollection/StudentComment.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (Howl404) [reply mw-webapp/src/model/firebaseCollection/StudentComment.ts]: Fully deleted this model

File: mw-webapp/src/model/firebaseCollection/DayReport.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (Howl404) [reply mw-webapp/src/model/firebaseCollection/DayReport.ts]: Fully deleted this model

File: mw-webapp/src/component/table/columns.tsx [hunk 9]
```diff
@@ -22,17 +22,24 @@ const getObjectArrayItem = (arrayItem: JobDone | PlanForNextPeriod | CurrentProb
   );
 };
 
-const getStringArrayItem = (arrayItem: string, index: string) => {
+
+/**
+ * Render a string item within a div element.
```
Comment (tritonJS826) [inline mw-webapp/src/component/table/columns.tsx:27]:
```suggestion
 * Render a string within a div element.
```

File: mw-webapp/src/component/table/columns.tsx [hunk 10]
```diff
@@ -22,17 +22,24 @@ const getObjectArrayItem = (arrayItem: JobDone | PlanForNextPeriod | CurrentProb
...
   return (
-    (!arrayItem) ?
-      <div />
-      :
-    //TODO: task #65 use flag instead of first index
-      <div key={index}>
-        <div className={arrayItem[INDEX_OF_CHECK_MARK] === "✓" ? styles.completed : styles.notCompleted}>
-          {arrayItem}
-        </div>
+    <div key={index}>
```
Comment (tritonJS826) [inline mw-webapp/src/component/table/columns.tsx:38]: index must not be used as key

File: mw-webapp/src/component/table/columns.tsx [hunk 11]
```diff
@@ -22,17 +22,24 @@ const getObjectArrayItem = (arrayItem: JobDone | PlanForNextPeriod | CurrentProb
...
+ * @param {boolean} [params.isDone] - Optional. If true, the item is styled as completed.
+ *
+ * @returns {JSX.Element} The rendered string item.
+ */
+const renderStringItem = ({text, index, isDone}: {text:string; index:string; isDone?: boolean}): JSX.Element => {
   return (
-    (!arrayItem) ?
-      <div />
-      :
-    //TODO: task #65 use flag instead of first index
```
Comment (tritonJS826) [inline mw-webapp/src/component/table/columns.tsx:30]: why did you remove this comment?

File: mw-webapp/src/component/table/columns.tsx [hunk 12]
```diff
@@ -56,7 +63,8 @@ const getDateValue = (cellValue: CellContext<DayReport, Date>) => {
 };
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (tritonJS826) [inline mw-webapp/src/component/table/columns.tsx:65]: do we need this line?

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts [hunk 13]
```diff
@@ -0,0 +1,32 @@
+import {MentorComment} from "src/model/businessModel/MentorComment";
+import {User} from "src/model/businessModel/User";
+import {MentorCommentDTO} from "src/model/firebaseCollection/MentorCommentDTO";
+import {UserDTO} from "src/model/firebaseCollection/UserDTO";
+
+export const MentorCommentDTOToMentorCommentConverter = (mentorComment: MentorCommentDTO, usersDTO: UserDTO[]) => {
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts:6]: check example and refacor it please

Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts:6]: looks like wrong approach, too much calculations inside. Do you really need to get all users for just one comment?

File: mw-webapp/src/dataAccessLogic/getDayReports.ts [hunk 14]
```diff
@@ -41,10 +44,18 @@ export const getDayReports = async (): Promise<DayReport[]> => {
       return problemForCurrentPeriod;
     });
 
+  const mentorComments = firstReport.mentorComments.
+    map((mentorCommentUuid) => {
+      const mentorComment: MentorComment = mentorCommentsPreview
+        .find((elem) => elem.uuid === mentorCommentUuid) ?? {} as MentorComment;
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/getDayReports.ts:50]: avoid as

File: mw-webapp/src/model/businessModel/DayReport.ts [hunk 15]
```diff
@@ -64,7 +63,8 @@ export class DayReport {
       new CurrentProblem(currentProblemItem));
     this.studentComments = dayReportData.studentComments;
     this.learnedForToday = dayReportData.learnedForToday;
-    this.mentorComments = dayReportData.mentorComments;
+    this.mentorComments = dayReportData.mentorComments?.map((mentorComment) =>
```
Comment (tritonJS826) [inline mw-webapp/src/model/businessModel/DayReport.ts:66]: why `?` is it possible. It is not javascript

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/table/columns.tsx (hunk 11 above)
Comment (Howl404) [reply mw-webapp/src/component/table/columns.tsx:30]: Because it's done

File: mw-webapp/src/component/table/columns.tsx (hunk 12 above)
Comment (Howl404) [reply mw-webapp/src/component/table/columns.tsx:65]: Removed

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts (hunk 13 above)
Comment (Howl404) [reply mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts:6]: Its even worse in next steps, i need to transfer all WaysDTO and assign them to User, i don't understand why we need User model instead of userUuid, it would be way easier

File: mw-webapp/src/dataAccessLogic/getDayReports.ts (hunk 14 above)
Comment (Howl404) [reply mw-webapp/src/dataAccessLogic/getDayReports.ts:50]: Don't know how to avoid it, added Error if mentor comment is not found, i dont know if its better approarch

File: mw-webapp/src/component/table/columns.tsx (hunk 10 above)
Comment (Howl404) [reply mw-webapp/src/component/table/columns.tsx:38]: Renamed parameter, in fact it's not a index

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts (hunk 13 above)
Comment (Howl404) [reply mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts:6]: I didnt understand which example and where

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts [hunk 16]
```diff
@@ -0,0 +1,30 @@
+import {MentorComment} from "src/model/businessModel/MentorComment";
+import {MentorCommentDTO} from "src/model/firebaseCollection/MentorCommentDTO";
+import {UserDTO} from "src/model/firebaseCollection/UserDTO";
+
+
```
Comment (Ekaterina1994) [inline mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts:5]: Delete empty line

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+  mentorCommentDTO: MentorCommentDTO, usersDTO: UserDTO[],
+): MentorComment => {
+  const matchingUser = usersDTO.find(user => user.uuid === mentorCommentDTO.mentorUuid);
+
+  if (!matchingUser) {
+    throw new Error(`User not found for uuid: ${mentorCommentDTO.mentorUuid}`);
+  }
+
+
+  return new MentorComment(
```
Comment (Ekaterina1994) [inline mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts:22]: Del empty line

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Convert {@link MentorCommentDTO} to {@link MentorComment}
+ * @param {MentorCommentDTO} mentorCommentDTO {@link MentorCommentDTO}
+ * @param {UserDTO[]} usersDTO - {@link UserDTO}
+ * @returns {MentorComment} mentorComment {@link MentorComment}
+ */
+export const MentorCommentDTOToMentorCommentConverter = (
+  mentorCommentDTO: MentorCommentDTO, usersDTO: UserDTO[],
+): MentorComment => {
+  const matchingUser = usersDTO.find(user => user.uuid === mentorCommentDTO.mentorUuid);
```
Comment (Ekaterina1994) [inline mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts:15]: Is i tpossible to write this in one line?

File: mw-webapp/src/dataAccessLogic/getDayReports.ts [hunk 19]
```diff
@@ -41,10 +43,21 @@ export const getDayReports = async (): Promise<DayReport[]> => {
       return problemForCurrentPeriod;
     });
 
+  const mentorComments = firstReport.mentorComments.
+    map((mentorCommentUuid) => {
+      const mentorComment = mentorCommentsPreview
```
Comment (Ekaterina1994) [inline mw-webapp/src/dataAccessLogic/getDayReports.ts:48]: one line. 
Check rules (eslint, prettier?)

File: mw-webapp/src/dataAccessLogic/getMentorComments.ts [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Mentor comments
+ * @returns {Promise<MentorComment[]>}
+ */
+export const getMentorComments = async (): Promise<MentorComment[]> => {
+  const mentorCommentsDTO = await MentorCommentsService.getMentorCommentsDTO();
+  const usersDTO = await UserService.getUsersDTO();
+  const mentorComments = mentorCommentsDTO.map((mentorComment) =>
+    (MentorCommentDTOToMentorCommentConverter(mentorComment, usersDTO)));
```
Comment (Ekaterina1994) [inline mw-webapp/src/dataAccessLogic/getMentorComments.ts:15]: one line or if line more them 130 symbols  make new line from .map

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts (hunk 18 above)
Comment (Howl404) [reply mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts:15]: 133 lenght, we have 130 max

File: mw-webapp/src/dataAccessLogic/getDayReports.ts (hunk 19 above)
Comment (Howl404) [reply mw-webapp/src/dataAccessLogic/getDayReports.ts:48]: Dont understand what you mean, just copied same style for other variable

File: mw-webapp/src/dataAccessLogic/getMentorComments.ts (hunk 20 above)
Comment (Howl404) [reply mw-webapp/src/dataAccessLogic/getMentorComments.ts]: Done

**Review by Ekaterina1994 — APPROVED** [review]

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts:14]: Why do you find user by uuid if you will never use other user properties except uuid?

File: mw-webapp/src/service/MentorCommentsService.ts [hunk 22]
```diff
@@ -0,0 +1,23 @@
+import {collection, getDocs} from "firebase/firestore";
+import {db} from "src/firebase";
+import {MentorCommentDTO} from "src/model/firebaseCollection/MentorCommentDTO";
+import {querySnapshotToDTOConverter} from "src/service/converter/querySnapshotToDTOConverter";
+
+const PATH_TO_MENTOR_COMMENTS_COLLECTION = "mentorComments";
+
+/**
+ * MentorComments requests: {@link getMentorComments}
```
Comment (tritonJS826) [inline mw-webapp/src/service/MentorCommentsService.ts:9]: 1. What do yo umean requests? Dont use synonims
2. what is  getMentorComments? How does it related to this service?

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/service/MentorCommentsService.ts (hunk 22 above)
Comment (Howl404) [reply mw-webapp/src/service/MentorCommentsService.ts:9]: I followed comment style from other services after their refactoring and moving to fireStore, i thought that's how they supposed to be

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts (hunk 21 above)
Comment (Howl404) [reply mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/mentorCommentDTOToMentorCommentConverter.ts:14]: Totally forgot about it after changing model, thank you, fixed

File: mw-webapp/src/service/MentorCommentsService.ts (hunk 22 above)
Comment (tritonJS826) [reply mw-webapp/src/service/MentorCommentsService.ts:9]: @link is a tool to ger reference in docs, that's all. You can tell everyone: if you can't add any additional info in the doc - that does not mean you should just type something using synonyms. Docs should simplify the project, not complexify)

Comment (Howl404) [reply mw-webapp/src/service/MentorCommentsService.ts]: So, should i change doc in other services too? Changed comment in this one, please check.

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/service/MentorCommentsService.ts (hunk 22 above)
Comment (tritonJS826) [reply mw-webapp/src/service/MentorCommentsService.ts]: It is up to you. Usually - yes if it is simle
